lib/mcstat.inc

```diff
@@ -98,47 +98,40 @@ function hr_time($seconds) {
  */
 class mcstat_client
 {
-  private $socket_ptr = null;
+  private $mc = null;
   private $hostname = null;
   private $port = null;
+  private $username = null;
+  private $password = null;
 
-  function __construct($hostname = DEFAULT_MEMCACHE_HOST, $port = DEFAULT_MEMCACHE_PORT) {
+  function __construct($hostname, $port, $username, $password) {
```

Comment:
Should username and password be optional params in order to preserve backward compatibility?

Reply:
Also, is there any reason we're dropping the default values for the first two params? It looks like those values are still being used as defaults in other places in this code.

Reply:
The params are validated and set to appropriate defaults in the entry point for the script which is `mcstat_session::main()` starting [here](https://github.com/webbj74/mcstat/blob/ca1a6fbf7ba3949353298277ffd9d8fe92558b2d/lib/mcstat.inc#L342) so I felt that also including them in this function declaration was redundant as these values should remain immutable for the duration of the script's run.

Additionally, one of the constants in the original default doesn't actually exist. `DEFAULT_MEMCACHE_HOST` should have been `DEFAULT_MEMCACHE_HOSTNAME`. 😄